Assets/Scripts/Models/Furniture.Factory.cs

```diff
@@ -0,0 +1,185 @@
+﻿using MoonSharp.Interpreter;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Xml.Serialization;
+
+/// <summary>
+/// InstalledObjects are things like walls, doors, and furniture (e.g. a sofa).
+/// </summary>
+public partial class Furniture : IXmlSerializable, ISelectable, IContextActionProvider
+{
+    // constants for parameters
+    const string CUR_PROCESSING_TIME_PARAM_NAME = "cur_processing_time";
+    const string MAX_PROCESSING_TIME_PARAM_NAME = "max_processing_time";
+    const string CUR_PROCESSED_INV_PARAM_NAME = "cur_processed_inv";
+    const string CUR_PRODUCTION_CHAIN_PARAM_NAME = "cur_production_chain";
+
+    private FactoryInfo factoryData;
+    
+    private List<FactoryContextMenu> factoryMenuActions;
+
+    private void UpdateFactory(float deltaTime)
+    {
+        // if there is enough input, do the processing and store item to output
+        // - remove items from input
+        // - add param to reflect factory can provide output (has output inside)
+        //   - as output will be produced after time, it is possible that output spot can be ocupied meanwhile
+        // - process for specified time
+        // - if output slot is free, provide output (if not, keep output 'inside' factory)
+
+        var curSetupChainName = furnParameters[CUR_PRODUCTION_CHAIN_PARAM_NAME].ToString();
+
+        if (!string.IsNullOrEmpty(curSetupChainName))
+        {
+            FactoryInfo.ProductionChain cChain = null;
```

Comment:
You should avoid prefixes on variables. If you want it really short, maybe `prodChain`? I'd name it `productionChain`.